extensions/amp-next-page/0.1/next-page-service.js

```diff
@@ -497,19 +515,59 @@ export class NextPageService {
 
   /**
    * Sets the specified document as active, updating the document title and URL.
-   * @param {!DocumentRef} documentRef Reference to the document to set as
-   *     active.
+   *
+   * @param {number} documentIndex Index of the document to be activated
    * @private
    */
-  setActiveDocument_(documentRef) {
-    const {amp} = documentRef;
-    this.win_.document.title = amp.title || '';
-    this.activeDocumentRef_ = documentRef;
-    this.setActiveDocumentInHistory_(documentRef);
+  setActiveDocument_(documentIndex) {
+    this.documentRefs_.forEach((docRef, index) => {
+      const {amp} = docRef;
+      let updatedVisibilityState;
+      // Update the title and history
+      if (index == documentIndex) {
+        this.win_.document.title = amp.title || '';
+        this.activeDocumentRef_ = docRef;
+        this.setActiveDocumentInHistory_(docRef);
+        updatedVisibilityState = VisibilityState.VISIBLE;
+      } else if (index !== 0) {
+        updatedVisibilityState = VisibilityState.HIDDEN;
+      }
+      // Show the active doc and hide other docs
+      if (updatedVisibilityState) {
+        this.setDocumentVisibility_(index, updatedVisibilityState);
```

Comment:
Would be shorter and a bit cleaner to just move this line to the two spots above and remove the `let` variable above.